### PR TITLE
Refactoring pieces

### PR DESCRIPTION
**Pull request by andersbogsnes**
Moved VizError to utils and inherit from main exception
Cleaned up docstrings
Cleaned up artifactory implementation
Cleaned up gridsearch implementation
Added ability to pass filename only to FileStorage.load

Added todos for next pass

- [x] Tests added
- [x] Passes flake8

### CODE REVIEW

File: src/ml_tooling/baseclass.py [hunk 1]
```diff
@@ -358,7 +362,8 @@ def score_estimator(
         metrics: Union[str, List[str]] = "default",
         cv: Optional[int] = False,
     ) -> "Result":
```
Comment (Valleo4) [inline src/ml_tooling/baseclass.py]:
```suggestion
    ) -> Result:
```

File: tests/test_artifactory.py [hunk 2]
```diff
@@ -1,49 +1,51 @@
+import pathlib
 import sys
 import pytest
 import joblib
-from unittest.mock import patch
+from unittest.mock import MagicMock
 
 from sklearn.pipeline import Pipeline
 from sklearn.base import BaseEstimator
```
Comment (Valleo4) [inline tests/test_artifactory.py]: from ml_tooling import Estimator

File: tests/test_artifactory.py [hunk 3]
```diff
@@ -1,49 +1,51 @@
...
 
-    mock_artifactory_path.return_value.deploy_file = mock_deploy_file
+    mock = MagicMock()
+    mock.__truediv__.deploy_file.return_value = mock_deploy_file()
 
     storage = ArtifactoryStorage("http://www.testy.com", "/test")
+    storage.artifactory_path = mock
     storage.save("test", "test")
     f = joblib.load(file_path)
     assert isinstance(f, (BaseEstimator, Pipeline))
```
Comment (Valleo4) [inline tests/test_artifactory.py]:
```suggestion
    assert isinstance(f, Estimator)
```

File: tests/test_artifactory.py [hunk 4]
```diff
@@ -1,49 +1,51 @@
...
+    mock_open = MagicMock()
+    mock_open.open.return_value.__enter__.return_value = open_estimator_pickle
+
+    mock_path = MagicMock()
+    mock_path.__truediv__.return_value.__truediv__.return_value = mock_open
+
+    storage = ArtifactoryStorage("testy", "test")
+    storage.artifactory_path = mock_path
+    f = storage.load("test")
     assert isinstance(f, (BaseEstimator, Pipeline))
```
Comment (Valleo4) [inline tests/test_artifactory.py]:
```suggestion
    assert isinstance(f, Estimator)
```

File: src/ml_tooling/storage/__init__.py [hunk 5]
```diff
@@ -1,4 +1,5 @@
 from .file import FileStorage
 from .base import Storage
+from .artifactory import ArtifactoryStorage
```
Comment (thomasfrederikhoeck) [inline src/ml_tooling/storage/__init__.py]: `ArtifactoryStorage` can't be in init. At least not with out try-catch statement as some users won't have it installed. That is why Travis is failing

File: tests/test_artifactory.py (hunk 3 above)
Comment (andersbogsnes) [reply tests/test_artifactory.py]: Unfortunately Typehints cannot be used as instance checks

File: src/ml_tooling/storage/__init__.py (hunk 5 above)
Comment (andersbogsnes) [reply src/ml_tooling/storage/__init__.py]: Changed to be lazy-loading - first throws exeception if artifactory is not installed when trying to instantiate Artifactory Storage

**Pull request merged**